Source/ThirdParty/scope_guard/scope_guard.hpp

```diff
@@ -0,0 +1,177 @@
+/*
+ *  Created on: 13/02/2018
+ *      Author: ricab
+ *
+ * See README.md for documentation of this header's public interface.
+ */
+
+#ifndef SCOPE_GUARD_HPP_
+#define SCOPE_GUARD_HPP_
+
+#include <type_traits>
+#include <utility>
+
+#if __cplusplus >= 201703L && defined(SG_REQUIRE_NOEXCEPT_IN_CPP17)
+#define SG_REQUIRE_NOEXCEPT
+#endif
+
+namespace sg
```

Comment:
I try to avoid implementations in this namespace and leave that to the end of the file. This way you can look at the code and see the signatures without being distracted by implementation details